### ISSUE TITLE
`docker rm -f` silently ignores errors and leads to inconsistent state

### ISSUE THREAD

**Issue by rade** [docker/docker#12825]
I've got some broken containers that docker refuses to remove...

````
$ docker ps -aq
3b784fa07fcd
d88aa2757378
2219bb22fe66
vagrant@host1:~$ docker rm 3b784fa07fcd
Error response from daemon: Cannot destroy container 3b784fa07fcd: Driver devicemapper failed to remove root filesystem 3b784fa07fcd4fab5bc8a09ef120f4784d103d8cabd20de5955120b0f94649f6: Device is Busy
FATA[0010] Error: failed to remove one or more containers 
````

So far so good, but now let's try removing that container with `rm -f`...

````
vagrant@host1:~$ docker rm -f 3b784fa07fcd
3b784fa07fcd
vagrant@host1:~$ echo $?
0
vagrant@host1:~$ docker ps -aq
d88aa2757378
2219bb22fe66
````
No error, and the container is gone. Well, that might be ok, except that when we bounce docker...

````
vagrant@host1:~$ sudo service docker restart
docker stop/waiting
docker start/running, process 1326
vagrant@host1:~$ docker ps -aq
3b784fa07fcd
d88aa2757378
2219bb22fe66
````
...the container comes back.

Version info:
````
$ docker version
Client version: 1.6.0
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 4749651
OS/Arch (client): linux/amd64
Server version: 1.6.0
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 4749651
OS/Arch (server): linux/amd64
````

**Comment by cpuguy83** [comment]
Thanks! Fixed in #12826.

**Comment by rade** [comment]
Wow, that was quick. Thanks!
